src/daemon/haven.rs

```diff
@@ -0,0 +1,63 @@
+use bytes::Bytes;
+use earendil_crypt::{Fingerprint, IdentityPublic, IdentitySecret};
+use earendil_packet::crypt::OnionPublic;
+use serde::{Deserialize, Serialize};
+use stdcode::StdcodeSerializeExt;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct HavenLocator {
```

Comment:
This is an example of a struct where it actually makes sense to expose all the fields as public :)